### PR TITLE
Branch1

### PR DESCRIPTION
**Pull request by ahmed-nad**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -10,28 +10,50 @@
...
 };
-struct Map
+class Map
 {
-	std::vector<Road> roads;
+	std::vector<Road> roads_;
+public:
+	void load_data_from_file(std::string file_name);
+	
+	float get_r1_l1_l_b_id(){return roads_[0].get_lanes()[0].get_left_boundary_id();}
```
Comment (ahmed-nad) [inline main.cpp]: please remove this

**Review by ahmed-nad — COMMENTED** [review]
1) Remove setters
2) Correct indentation
3) Use constructor initialization

File: main.cpp (hunk 1 above)
Comment (hamzaabbasi16) [reply main.cpp]: Removed!

**Review by ahmed-nad — COMMENTED** [review]
Divide the classes into separate .h, .cpp 

Default constructor is available

File: main.cpp [hunk 2]
```diff
@@ -7,126 +7,14 @@
...
-		vec.push_back(json_point.get<array>()[0].get<double>());
-		vec.push_back(json_point.get<array>()[1].get<double>());
-		//std::cout<<"x: "<<vec[0]<<", y: "<<vec[1]<<std::endl;
-		lane_4.centre_point.push_back(vec);
-	}
 
-	// ********** Loading the lanes to the road structure ***********
+#include "Lane.h"
+#include "Road.h"
+#include "Map.h"
```
Comment (ahmed-nad) [inline main.cpp]: you only need to include map here

**Review by ahmed-nad — COMMENTED** [review]
Divide the classes into .h and .cpp files separately please

**Review by ahmed-nad — COMMENTED** [review]
Also please upload the cmake file as well

File: Lane.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	float get_left_boundary_id(){return left_boundary_id;}
+	float get_right_boundary_id(){return right_boundary_id;}
+	std::vector<std::vector<double>> get_centre_point(){return centre_point;}
+};
+Lane :: Lane(float a, float b, float c, std::vector<std::vector<double>> d)
+{
+	id = a;
+	left_boundary_id = b;
+	right_boundary_id = c;
+	centre_point = d;
```
Comment (ahmed-nad) [inline Lane.h]: you can remove these default constructors as they are not necessary

**Review by ahmed-nad — COMMENTED** [review]
remove default constructors as well they are not necessary

File: Lane.h (hunk 3 above)
Comment (hamzaabbasi16) [reply Lane.h]: then how'd we initialize the data while instantiating the object of Lane and Road?

**Pull request merged**